### ISSUE TITLE
Tests fail with pytest 5

### ISSUE THREAD

**Issue by scarabeusiv** [python-hyper/h11#86]
```
[    4s] >       assert "kwarg a" in str(exc)
[    4s] E       AssertionError: assert 'kwarg a' in '<ExceptionInfo TypeError tblen=2>'
[    4s] E        +  where '<ExceptionInfo TypeError tblen=2>' = str(<ExceptionInfo TypeError tblen=2>)
[    4s] 
[    4s] h11/tests/test_events.py:47: AssertionError
```

The syntax bit changed and you just can't do exception but instead have to call exception.value,

**Comment by pquentin** [comment]
Hi! Thanks for reporting that issue.

So that's an interesting situation:

* h11 can't use pytest 5 because it supports Python 2, so we actually need to specify `pytest<5` in the test requirements.
* However, I believe exc.value is the recommended form in pytest 4 too, so fixing that would be great

Would you be interested in putting together a pull request with those two changes?

Thanks

**Comment by scarabeusiv** [comment]
You can use pytest 5 for python3 only, even in travis; in short it will pick pytest 4 on python2 as there is nothing newer and pytest 5.1.1 (atm) for the python3 always keeping it on the latest release.

There might be minor issue if the two start to diverge too much and ie using pytest 5.2 would break up the behaviour with pytest 4 completely, but I suppose the upstream pytest devs are not that crazy :)

**Comment by pquentin** [comment]
That's an option too, yes

**Comment by njsmith** [comment]
Fixed by #87
